pkg/sessions/redis/session_store_test.go

```diff
@@ -0,0 +1,85 @@
+package redis
+
+import (
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/Bose/minisentinel"
+	"github.com/alicebob/miniredis/v2"
+	"github.com/go-redis/redis/v7"
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/apis/options"
+	sessionsapi "github.com/oauth2-proxy/oauth2-proxy/pkg/apis/sessions"
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/logger"
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/sessions/tests"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestSessionStore(t *testing.T) {
+	logger.SetOutput(GinkgoWriter)
+
+	redisLogger := log.New(os.Stderr, "redis: ", log.LstdFlags|log.Lshortfile)
+	redisLogger.SetOutput(GinkgoWriter)
+	redis.SetLogger(redisLogger)
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Redis SessionStore")
+}
+
+var _ = Describe("Redis SessionStore Tests", func() {
+	var mr *miniredis.Miniredis
+
+	BeforeEach(func() {
+		var err error
+		mr, err = miniredis.Run()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		mr.Close()
+	})
+
+	tests.RunSessionStoreTests(
```

Comment:
We support redis cluster in our options, but I don't see a test case for that.  Probably out of scope for this PR, but maybe something to raise an issue about for us to take care of in the future.

With a quick glance, it looks like `miniredis` at least has support for the subset of cluster commands that redis cluster mode added.

Reply:
Nice -- glad it wasn't too bad to add in!